### PR TITLE
[JENKINS-50291] Customize where Async[A]PeriodicWork tasks are logged

### PR DESCRIPTION
**Pull request by batmat**
See [JENKINS-50291](https://issues.jenkins-ci.org/browse/JENKINS-50291).

### Proposed changelog entries

* [JENKINS-50291](https://issues.jenkins-ci.org/browse/JENKINS-50291): Introduce system property to send logs usually sent to `$JENKINS_HOME/logs` to another location. 

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/triggers/SafeTimerTask.java
Comment (batmat) [inline core/src/main/java/hudson/triggers/SafeTimerTask.java]: /me ponders: this sysprop name is not very consistent with `getLogsRoot()` intent, though it's the case for now, it's not only for **tasks** logging...

**Comment by batmat** [comment]
@oleg-nenashev I know you're very busy, but I requested your review as it seems this PR at least relates to the [External Build Log storage for Jenkins Epic|https://issues.jenkins-ci.org/browse/JENKINS-38313]. Do you think it could either get in the way of it, or be advantagely superseded by using the outcome there?

Note that I would also be happy to have the current PR as a way to move forward wrt https://github.com/jenkinsci/jep/tree/master/jep/302#data_segregation and refactor later to use JENKINS-38313 once completed.

cc @jglick who pointed this out.

File: core/src/main/java/hudson/triggers/SafeTimerTask.java
Comment (oleg-nenashev) [reply core/src/main/java/hudson/triggers/SafeTimerTask.java]: Yes ".logsTargetDir" would be better. Maybe it also makes sense to move the property to another class. "SafeTimerTask" barely has any relation to logging for everybody excepting Jenkins core devs.

**Review by oleg-nenashev — APPROVED** [review]
small 🐛 , just for the property naming.

LGTM from the implementation PoV. I think the property/method could be moved to another class so that the property name is user-friendly.

Other concern: Support Core plugin will stop collecting the data if the property is set. I would recommend creating a downstream pull request, which makes the plugin aware about the property so that it can handle it correctly.

**Comment by batmat** [comment]
> move the property to another class.

Actually, it took me some time to choose, since I wanted it to be effective both for `AsyncPeriodicWork` and `AsyncAperiodicWork`, and this was the only common parent, so I thought I made some sense. 

The only way out for this seemed to me to create a new class dedicated to it... Like `JenkinsWork` or so (yeah, naming things :D).

Anyway thanks a lot for the review. Will see how to address your comment.
Ack also about _support-core plugin_.

File: core/src/main/java/hudson/triggers/SafeTimerTask.java
Comment (batmat) [reply core/src/main/java/hudson/triggers/SafeTimerTask.java]: @oleg-nenashev fixed. Didn't move to another class though, since I'm not sure which one to use and hence that it would really be better. Thanks again for the review!

**Comment by batmat** [comment]
Will merge later today if nobody objects.

**Review by oleg-nenashev — APPROVED** [review]
https://wiki.jenkins.io/display/JENKINS/Features+controlled+by+system+properties will need to be updated

**Comment by batmat** [comment]
Yes :+1: , I had already put that in the proposed changelog to remember:
image

**Comment by batmat** [comment]
Done https://wiki.jenkins.io/display/JENKINS/Features+controlled+by+system+properties?a=blah

**Pull request merged**
